### ISSUE TITLE
Abstract Pandora user interactions.

### ISSUE THREAD

**Issue by arbales** [nixme/warble#14]
Currently, you're given a limited view of your Pandora stations. We're showing the current state of the station. Instead, we could simply allow a user to start a station, and periodically pull from the station. To emphasize the ever-building local cache of the station, the sourcelist will show a mixture of recent pulls and previously cached content — this will also reduce our Pandora API load as time progresses.

This way, we can move to a more abstracted music display model, that allows us to view the caches by artist, album, source. Over time, this will help reduce the barrier to adding new services.

**Comment by nixme** [comment]
This would be great but would require a bit of re-architecture on how we store songs and grab them. I'll keep it in mind when I refactor the backend a bit this week.

**Comment by arbales** [comment]
Was thinking about this — it'd be cool if we could eventually switch to a datastore thats a little more relational. If we wanted to automagically fill the queue from things like a list of users or other factors, we'd want to be able to run joins.

**Comment by nixme** [comment]
Yes, Redis was unnecessary and overkill. I just wanted to play with it. PostgreSQL or even sqlite are better choices.

**Comment by nixme** [comment]
The backend is closer to getting this sort of functionality implemented. master now uses PostgreSQL for storing everything. Redis is just used for the playlist, background worker queues, and pub-sub to socket.io.
